controller/controller.go

```diff
@@ -53,14 +64,13 @@ func (c *Controller) GetDefaultGatewayIp(net *types.VirtualNetwork) (string, err
 }
 
 func (c *Controller) GetOrCreateInstance(tenantName, containerId string) (*types.VirtualMachine, error) {
-	name := fmt.Sprintf("%s:%s:%s", common.DomainName, tenantName, containerId)
-	instance, err := types.VirtualMachineByName(c.ApiClient, name)
+	instance, err := types.VirtualMachineByName(c.ApiClient, containerId)
 	if err == nil && instance != nil {
```

Comment:
err == nil && instance != nil means that an existing machine was found - so we return it. Else, we create a new one